pkgs/development/tools/build-managers/apache-maven/build-package.nix

```diff
@@ -26,13 +29,28 @@ let
       maven
     ];
 
-    buildPhase = ''
-      runHook preBuild
+    buildPhase =
+      if buildOffline
+      then ''
+        runHook preBuild
 
-      mvn package -Dmaven.repo.local=$out/.m2 ${mvnParameters}
+        mvn dependency:go-offline -Dmaven.repo.local=$out/.m2 ${mvnDepsParameters}
```

Comment:
We might consider dropping the old `buildPhase` if it works for all packages. `mvnHash` should however be updated globally.

Reply:
@wegank Unfortunately, it does not work for all packages. I already tested this with dbeaver which fails:
https://github.com/dbeaver/dbeaver/issues/21217.

As a result, I used a feature flag and the `if ... then ''..'' else ''..''` syntax.